tests/model_registry/scc/test_model_registry_scc.py

```diff
@@ -0,0 +1,144 @@
+    [
+        pytest.param(
+            {"namespace_name": MR_NAMESPACE},
+            {
+                "component_patch": {
+                    DscComponents.MODELREGISTRY: {
+                        "managementState": DscComponents.ManagementState.MANAGED,
+                        "registriesNamespace": MR_NAMESPACE,
+                    },
+                },
+            },
+            MODEL_DICT,
+        ),
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures("model_registry_namespace", "updated_dsc_component_state_scope_class", "registered_model")
+class TestModelRegistrySecurityContextValidation:
+    @pytest.mark.parametrize(
+        "model_registry_resource",
+        [
+            pytest.param({"kind": Deployment}),
+        ],
+        indirect=["model_registry_resource"],
+    )
+    def test_model_registry_deployment_security_context_validation(
+        self: Self,
+        model_registry_resource: NamespacedResource,
+        model_registry_namespace: Namespace,
+    ):
+        """
+        Validate that model registry deployment does not set runAsUser/runAsGroup
+        """
+        error = []
+        for container in model_registry_resource.instance.spec.template.spec.containers:
+            if not all([True for key in KEYS_TO_VALIDATE if not container.get(key)]):
+                error.append({container.name: container.securityContext})
+
+        if error:
+            pytest.fail(
+                f"{model_registry_resource.name} {model_registry_resource.kind} containers expected to not "
+                f"set {KEYS_TO_VALIDATE}, actual: {error}"
+            )
+
+    @pytest.mark.parametrize(
+        "model_registry_resource",
+        [
+            pytest.param({"kind": Pod}),
+        ],
+        indirect=["model_registry_resource"],
+    )
+    def test_model_registry_pod_security_context_validation(
+        self: Self,
+        model_registry_resource: NamespacedResource,
+        model_registry_scc_namespace: dict[str, str],
+    ):
+        """
+        Validate that model registry pod gets runAsUser/runAsGroup from openshift and the values matches namespace
+        annotations
+        """
+        errors = []
+        ns_uid = get_uid_from_namespace(namespace_scc=model_registry_scc_namespace)
+        pod_spec = model_registry_resource.instance.spec
+        errors = validate_pod_security_context(
+            pod_security_context=pod_spec.securityContext,
+            namespace_scc=model_registry_scc_namespace,
+            model_registry_resource=model_registry_resource,
+            ns_uid=ns_uid,
+        )
+        containers = model_registry_resource.instance.spec.containers
+        for container in containers:
+            expected_value = {
+                "runAsUser": int(ns_uid) + 1 if "sidecar" in container.args else int(ns_uid),
+                "runAsGroup": int(ns_uid) + 1 if "sidecar" in container.args else None,
+            }
```

Comment:
Are we sure these will always be the expected values? I thought it might be better to check that whatever value is present sits in the range defined in the namespace annotation

Reply:
OSSM explicitly sets runAsUser to be app_UID+1, and runAsGroup app_UID+1, https://github.com/maistra/istio/pull/1057. For other container, openshift sets it to app_UID. So this should not be flaky.

Reply:
Are you sure our uids are set by OSSM and not by OCP directly?

Reply:
yes. Proxy one comes from OSSM.